### ISSUE TITLE
Allow aria2c to preserve directory structure

### ISSUE THREAD

**Issue by MCOfficer** [KoalaBear84/OpenDirectoryDownloader#46]
**Is your feature request related to a problem? Please describe.**
Just noticed that according to the README, aria2c does not preserve directory structure while downloading.

**Describe the solution you'd like**
This is in fact possible by formatting the URLs file differently (See https://stackoverflow.com/a/28553355/7653274). This feature would best be hidden behind a flag, because it does make the file wget-incompatible.

**Comment by KoalaBear84** [comment]
True, first I thought it did, so I added it to ODD README, later I saw it (by default) doesn't support it, so I moved it as the second option and added the warning. It's sad that it doesn't do this, as it is very easy to add for them for the standard structure..

Hopefully I can add it. Probably must be a separate command line parameter which will create a second file?

**Comment by MCOfficer** [comment]
- parameter that creates a `*-aria2c.txt` file

- always create a `*-aria2c.txt` file

- parameter that makes the normal .txt file aria2c-compatible


pick one ¯\\_(ツ)_/¯

**Comment by Chaphasilor** [comment]
For what it's worth, I'm voting for the first option 😁

Second option seems unnecessary and the third option basically renders the URL file useless for most other things...

**Comment by KoalaBear84** [comment]
It will be better to focus on #20 because of the limited time.

**Comment by Chaphasilor** [comment]
That's what I thought, yeah :D

**Comment by ztgasdf** [comment]
Bumping as #20 got closed due to metalink becoming essentially defunct.

**Comment by KoalaBear84** [comment]
Yes, let's reopen it. Might not be super easy, writing the file of course is, but determining the directory might not be, the file should be easy as that is already parsed as a filename.

Also need to take care about special characters, which are different in Windows and Unix, but that may be taken care of in Aria2.

**Comment by maaaaz** [comment]
This asked feature is really important as `wget` supports the `keep directories` function with the `-x` option ; but wget is not as fast as `aria2c` or `axel` (both are faster thanks to parallel multiple-segment download).

We really need ODD to allow to download fast while keeping directory structure :)

**Comment by KoalaBear84** [comment]
Finally! Only tested with 'normal' directories. Maybe it fails (in aria2) which some special URL formats.

https://github.com/KoalaBear84/OpenDirectoryDownloader/releases/tag/v3.4.0.0
